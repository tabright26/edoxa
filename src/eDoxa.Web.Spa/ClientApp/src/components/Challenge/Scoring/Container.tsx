import { connect, MapStateToProps } from "react-redux";
import Scoring from "./Scoring";
import { RootState } from "store/types";
import { RouteComponentProps, withRouter } from "react-router-dom";
import { ChallengeId, ChallengeScoring } from "types";
import { compose } from "recompose";

interface Params {
  readonly challengeId: ChallengeId;
}

type OwnProps = RouteComponentProps<Params>;

interface StateProps {
  readonly scoring: ChallengeScoring;
}

const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (state, ownProps) => {
  const { data } = state.root.challenge;
  const challenge = data.find(challenge => challenge.id === ownProps.match.params.challengeId);
  return {
    scoring: challenge.scoring
  };
};

const enhance = compose<any, any>(
  withRouter,
  connect(mapStateToProps)
);

export default enhance(Scoring);