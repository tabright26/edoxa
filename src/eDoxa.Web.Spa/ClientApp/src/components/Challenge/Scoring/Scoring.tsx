import React, { FunctionComponent } from "react";
import { Card, CardHeader, Table } from "reactstrap";
import { ChallengeScoring, ChallengeId } from "types";
import { sentenceCase } from "change-case";

interface Props {
  readonly challengeId: ChallengeId;
  readonly scoring: ChallengeScoring;
  readonly className?: string;
}

const Scoring: FunctionComponent<Props> = ({ scoring, className }) => (
  <Card className={className}>
    <CardHeader className="text-center bg-gray-900">
      <strong className="text-uppercase">Scoring legend</strong>
    </CardHeader>
    <Table striped bordered hover size="sm" className="m-0">
      <tbody>
        {Object.entries(scoring).map((item, index) => (
          <tr key={index}>
            <td className="pl-3 align-middle">{sentenceCase(item[0])}</td>
            <td className="pr-3 align-middle text-right">{item[1]}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  </Card>
);

export default Scoring;
