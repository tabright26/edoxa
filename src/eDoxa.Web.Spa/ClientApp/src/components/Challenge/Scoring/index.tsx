import React, { FunctionComponent } from "react";
import { Card, CardHeader, Table } from "reactstrap";
import { ChallengeScoring, ChallengeId } from "types";
import { sentenceCase } from "change-case";
import { connect, MapStateToProps } from "react-redux";
import { RootState } from "store/types";
import { RouteComponentProps, withRouter } from "react-router-dom";
import { compose } from "recompose";

type Params = {
  readonly challengeId?: ChallengeId;
};

type OwnProps = RouteComponentProps<Params> & Params;

type StateProps = {
  readonly scoring: ChallengeScoring;
};

type InnerProps = OwnProps & StateProps;

type OutterProps = Params & { readonly className?: string };

type Props = InnerProps & OutterProps;

const Scoring: FunctionComponent<Props> = ({ scoring, className }) => (
  <Card className={className}>
    <CardHeader className="text-center bg-gray-900">
      <strong className="text-uppercase">Scoring legend</strong>
    </CardHeader>
    <Table striped bordered hover size="sm" className="m-0">
      <tbody>
        {Object.entries(scoring).map((item, index) => (
          <tr key={index}>
            <td className="pl-3 align-middle w-75">{sentenceCase(item[0])}</td>
            <td className="pr-3 align-middle w-25 text-right">{item[1]}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  </Card>
);

const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (
  state,
  ownProps
) => {
  const { data } = state.root.challenge;
  const challenge = data.find(
    challenge =>
      challenge.id ===
      (ownProps.match
        ? ownProps.match.params.challengeId
        : ownProps.challengeId)
  );
  return {
    scoring: challenge.scoring
  };
};

const enhance = compose<InnerProps, OutterProps>(
  withRouter,
  connect(mapStateToProps)
);

export default enhance(Scoring);
