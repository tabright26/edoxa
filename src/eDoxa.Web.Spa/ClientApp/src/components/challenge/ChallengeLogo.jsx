import React from 'react';
import { Card } from 'react-bootstrap';

import logo from '../../assets/images/logos/LeagueOfLegends.png';

const ChallengeLogo = () => (
  <Card className="bg-dark mx-3 my-4">
    <Card.Img className="p-4" src={logo} />
  </Card>
);

export default ChallengeLogo;
