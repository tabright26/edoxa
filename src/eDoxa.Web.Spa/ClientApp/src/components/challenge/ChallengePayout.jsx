import React from 'react';
import { Table, Card } from 'react-bootstrap';

import Spinner from '../Shared/Spinner';
import CurrencyFormat from '../Shared/Formaters/CurrencyFormat';
import BucketFormat from '../Shared/Formaters/BucketFormat';

const Render = ({ challenge }) => {
  if (!challenge) {
    return (
      <Card.Body>
        <Spinner />
      </Card.Body>
    );
  } else {
    let prevSize = 1;
    let nextSize = 0;
    return (
      <Table striped bordered hover size="sm" variant="dark" className="m-0">
        <thead>
          <tr>
            <th>Position</th>
            <th>Prize</th>
          </tr>
        </thead>
        <tbody>
          {/* {challenge.payout.buckets.map((bucket, index) => {
            const bucketSize = bucket.size;
            nextSize += bucketSize;
            const bucketRender = (
              <tr key={index}>
                <td>
                  { <BucketFormat prevSize={prevSize} nextSize={nextSize} /> }
                </td>
                <td>
                  { <CurrencyFormat
                    currency={challenge.payout.prizePool.currency}
                    amount={bucket.prize}
                  /> }
                </td>
              </tr>
            );
            prevSize += bucketSize;
            return bucketRender;
          })} */}
        </tbody>
      </Table>
    );
  }
};

const ChallengePayout = ({ challenge }) => {
  return (
    <Card bg="dark" className="my-4 text-light text-center">
      <Card.Header as="h5">
        {/* Payout ({challenge ? challenge.payout.prizePool.currency : ''}) */}
      </Card.Header>
      <Render challenge={challenge} />
    </Card>
  );
};

export default ChallengePayout;
