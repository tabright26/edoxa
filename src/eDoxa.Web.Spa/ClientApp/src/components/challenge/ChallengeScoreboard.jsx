import React from 'react';
import { Card, Accordion } from 'react-bootstrap';

import Scrollbar from 'react-scrollbars-custom';

import Participants from './Participants';

const ChallengeScoreboard = ({ challenge }) => (
  <>
    <Card bg="dark" className="my-2 text-light text-center">
      <Card.Header as="h5" className="border-0">
        Scoreboard
      </Card.Header>
    </Card>
    <Scrollbar
      style={{
        height: '500px'
      }}
    >
      <Accordion>
        <Participants challenge={challenge} />
      </Accordion>
    </Scrollbar>
  </>
);

export default ChallengeScoreboard;
