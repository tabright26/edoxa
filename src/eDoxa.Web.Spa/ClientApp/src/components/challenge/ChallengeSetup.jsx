import React from 'react';
import { Card, Row, Col, Badge, ProgressBar } from 'react-bootstrap';

import CurrencyFormat from '../Shared/Formaters/CurrencyFormat';
import Spinner from '../Shared/Spinner';

import ChallengeTimeline from './ChallengeTimeline';

import faker from 'faker';

faker.seed(1);

const Body = ({ challenge }) => {
  if (!challenge) {
    return (
      <Card.Body className="text-center text-white">
        <Spinner />
      </Card.Body>
    );
  } else {
    return (
      <Card.Body className="d-flex">
        <ChallengeTimeline challenge={challenge} />
        <Row>
          <Col>
            <Card.Title as="h3">Challenge setup</Card.Title>
            <hr className="border-light" />
            <dl className="row mb-0 float-right">
              <dt className="col-5">Name</dt>
              <dd className="col-7">{challenge.name}</dd>
              <dt className="col-5">Game</dt>
              <dd className="col-7">
                <Badge variant="primary">{challenge.game}</Badge>
              </dd>
              <dt className="col-5">
                {/* Entry fee ({challenge.setup.entryFee.currency}) */}
              </dt>
              <dd className="col-7">
                {/* <CurrencyFormat
                  currency={challenge.setup.entryFee.currency}
                  amount={challenge.setup.entryFee.amount}
                /> */}
              </dd>
              <dt className="col-5">Entries</dt>
              <dd className="col-7 my-auto">
                <ProgressBar
                  variant="primary"
                  now={challenge.participants.length}
                  max={challenge.entries}
                  label={`${challenge.participants.length}/${challenge.entries}`}
                />
              </dd>
              <dt className="col-5 mt-2 mb-1">Payout entries</dt>
              <dd className="col-7 mt-2 mb-1">
                {/* {challenge.setup.payoutEntries} */}
              </dd>
              <dt className="col-5 mt-1">Best of</dt>
              <dd className="col-7 mt-1 mb-0">{challenge.bestOf}</dd>
            </dl>
          </Col>
        </Row>
      </Card.Body>
    );
  }
};

const ChallengeSetup = ({ challenge }) => (
  <Card bg="dark" className="my-4 text-light">
    <Body challenge={challenge} />
  </Card>
);

export default ChallengeSetup;
