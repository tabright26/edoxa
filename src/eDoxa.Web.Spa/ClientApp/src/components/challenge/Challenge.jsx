import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container, Row, Col } from 'react-bootstrap';

import ChallengeLogo from './ChallengeLogo';
import ChallengeSetup from './ChallengeSetup';
import ChallengeScoreboard from './ChallengeScoreboard';
import ChallengeScoring from './ChallengeScoring';
import ChallengePayout from './ChallengePayout';

import faker from 'faker';

import { findChallenge } from '../../store/actions/arenaChallengeActions';

faker.seed(1);

class Challenge extends Component {
  componentDidMount() {
    // Check if the challenge exists in the store.
    if (!this.props.challenge) {
      // If the challenge does not exist, dispatch a redux action to find the challenge with the API call.
      // This call will update the store with the HTTP response call.
      this.props.actions.findChallenge();
    }
  }

  render() {
    return (
      <Container fluid={true}>
        <Row>
          <ChallengeLogo />
          <Col xs={4} className="ml-4">
            <ChallengeSetup challenge={this.props.challenge} />
          </Col>
        </Row>
        <Row>
          <Col xs={10}>
            <ChallengeScoreboard challenge={this.props.challenge} />
          </Col>
          <Col xs={2}>
            <Row>
              <Col xs={12}>
                <ChallengeScoring challenge={this.props.challenge} />
              </Col>
              <Col xs={12}>
                <ChallengePayout challenge={this.props.challenge} />
              </Col>
            </Row>
          </Col>
        </Row>
      </Container>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    challenge: state.arena.challenges.find(
      challenge => challenge.id === ownProps.match.params.challengeId
    )
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    actions: {
      findChallenge: () =>
        dispatch(findChallenge(ownProps.match.params.challengeId))
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Challenge);
