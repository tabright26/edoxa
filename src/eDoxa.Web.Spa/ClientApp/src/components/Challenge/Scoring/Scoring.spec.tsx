import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import Scoring from "./Scoring";
import { ChallengesState } from "store/root/challenge/types";

it("renders without crashing", () => {
  //Arrange
  const challenges: ChallengesState = {
    data: [
      {
        id: "123",
        name: "",
        state: "Inscription",
        bestOf: 3,
        entries: 10,
        game: "LeagueOfLegends",
        payoutEntries: 5,
        entryFee: { currency: "token", amount: 0 },
        participants: [],
        timeline: {
          createdAt: 123123123,
          startedAt: null,
          endedAt: null,
          closedAt: null
        },
        scoring: new Map<string, string>(),
        payout: {
          prizePool: { currency: "token", amount: 200000 },
          buckets: []
        }
      },
      {
        id: "456",
        name: "",
        state: "Inscription",
        bestOf: 3,
        entries: 10,
        game: "LeagueOfLegends",
        payoutEntries: 5,
        entryFee: { currency: "token", amount: 0 },
        participants: [],
        timeline: {
          createdAt: 123123123,
          startedAt: null,
          endedAt: null,
          closedAt: null
        },
        scoring: new Map<string, string>(),
        payout: {
          prizePool: { currency: "token", amount: 200000 },
          buckets: []
        }
      },
      {
        id: "678",
        name: "",
        state: "Inscription",
        bestOf: 3,
        entries: 10,
        game: "LeagueOfLegends",
        payoutEntries: 5,
        entryFee: { currency: "token", amount: 0 },
        participants: [],
        timeline: {
          createdAt: 123123123,
          startedAt: null,
          endedAt: null,
          closedAt: null
        },
        scoring: new Map<string, string>(),
        payout: {
          prizePool: { currency: "token", amount: 200000 },
          buckets: []
        }
      }
    ],
    loading: false,
    error: null
  };

  const store: any = {
    getState: () => {
      return {
        root: {
          challenge: challenges
        }
      };
    },
    dispatch: action => {},
    subscribe: () => {}
  };

  // Arrange
  const scoring = new Map<string, string>();

  //Act
  const tree = renderer.create(<Scoring scoring={scoring} />).toJSON();

  //Assert
  expect(tree).toMatchSnapshot();
});
