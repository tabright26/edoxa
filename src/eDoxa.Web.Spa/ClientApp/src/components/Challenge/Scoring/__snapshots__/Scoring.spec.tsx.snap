// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="card"
>
  <div
    className="text-center bg-gray-900 card-header"
  >
    <strong
      className="text-uppercase"
    >
      Scoring legend
    </strong>
  </div>
  <table
    className="m-0 table table-sm table-bordered table-striped table-hover"
  >
    <tbody />
  </table>
</div>
`;
